tests/MongoDB.Driver.Core.Tests/Core/ConnectionPools/ExclusiveConnectionPoolTests.cs

```diff
@@ -117,6 +128,11 @@ public void Constructor_should_throw_when_eventSubscriber_is_null()
             }
 
             act.ShouldThrow<InvalidOperationException>();
+            _capturedEvents.Next(); // skip ConnectionPoolCheckingOutConnectionEvent
```

Comment:
It might be a good idea to assert that the event we are skipping is in fact of the expected type:

    _capturedEvents.Next().Should().BeOfType<ConnectionPoolCheckingOutConnectionEvent>(); // skip ConnectionPoolCheckingOutConnectionEvent

Reply:
Same for other places where you've added code to skip events.

Reply:
done